ARCHITECTURE.md

```diff
@@ -1,113 +1,280 @@
 
-Once initialized successfully, the in-game PlayerAvatar GameObject hierarchy inside `BossRoom` scene will look something like (in the case of a selected Archer Boy class):
+Inside `CharSelect` scene, clients select from 8 possible avatar classes, and that selection is stored inside PersistentPlayer's `NetworkAvatarGuidState`.
+
+Inside `BossRoom` scene, `ServerBossRoomState` spawns a `PlayerAvatar` per PersistentPlayer present.
+
+Once initialized successfully, the `PlayerAvatar` GameObject hierarchy inside `BossRoom` scene will look something like (in the case of a selected Archer Boy class):
 
-* Player Avatar: a `NetworkObject` that *will* be destroyed when `BossRoom` scene is unloaded
-  * Player Graphics: a child GameObject containing `NetworkAnimator` component responsible for replicating animations invoked on the server
+* PlayerAvatar: a `NetworkObject` that will be destroyed when `BossRoom` scene is unloaded
+  * PlayerGraphics: a child GameObject containing `NetworkAnimator` component responsible for replicating animations invoked on the server
     * PlayerGraphics_Archer_Boy: a purely graphical representation of the selected avatar class
+
+`ClientAvatarGuidHandler`, a `NetworkBehaviour` component residing on the `PlayerAvatar` prefab instance will fetch the validated avatar GUID from `NetworkAvatarGuidState`, and spawn a local, non-networked graphics GameObject corresponding to the avatar GUID.
+
+### Characters
+
+`ServerCharacter` lives on a PlayerAvatar or other NPC character and contains server RPCs and NetworkVariables that store the state of any given character. It is responsible for executing or kicking off the server-side logic for the characters, which includes:
+
+- movement and pathfinding via `ServerCharacterMovement` - it uses NavMeshAgent that lives on the server to translate the character’s transform, which is synchronized using NetworkTransform component;
+- player action queueing and execution via `ServerActionPlayer`;
+- AI logic via `AIBrain` (applies to NPCs);
+- Character animations via `ServerAnimationHandler`, which themselves are synchronized using NetworkAnimator;
+
+`ClientCharacter` primarily is a host for the `ClientActionPlayer` class, and it also contains the client RPCs for the character gameplay logic.
+
+### Game config setup
+
+Game config in Boss Room is defined in `ScriptableObjects`.
+
+A singleton class [GameDataSource.cs](Assets/Scripts/Gameplay/GameplayObjects/RuntimeDataContainers/GameDataSource.cs) is responsible for storing all the actions and character classes in the game.
+
+[CharacterClass](Assets/Scripts/Gameplay/Configuration/CharacterClass.cs) is the data representation of a Character, containing such things as its starting stats and a list of Actions that it can perform. This covers both player characters and NPCs alike.
```

Comment:
```suggestion
[CharacterClass](Assets/Scripts/Gameplay/Configuration/CharacterClass.cs) is the data representation of a Character, containing elements such as starting stats and a list of Actions that it can perform. This covers both player characters and NPCs alike.
```